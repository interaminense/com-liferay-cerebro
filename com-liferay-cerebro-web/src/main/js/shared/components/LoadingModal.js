import Component, {Config} from 'metal-jsx';

import Icon from 'shared/components/Icon';
import Modal from './modal';
import Spinner from './Spinner';

class LoadingModal extends Component {
	render() {
		const {icon, message, title} = this.props;

		return (
			<Modal elementClasses="loading-modal-root" size="sm">
				{title && <h1 class="title">{title}</h1>}

				<div class="icon-container">
					{
						do {
							if (icon) {
								<Icon size="xl" symbol={icon} />;
							} else {
								<Spinner size="lg" />;
							}
						}
					}
				</div>

				{message && <p class="message">{message}</p>}
			</Modal>
		);
	}
}

LoadingModal.PROPS = {
	icon: Config.string(),
	message: Config.string().value(Liferay.Language.get('loading')),
	title: Config.string()
};

export default LoadingModal;