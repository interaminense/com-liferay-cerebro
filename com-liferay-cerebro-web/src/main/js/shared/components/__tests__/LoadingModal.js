import LoadingModal from '../LoadingModal';

describe('LoadingModal', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new LoadingModal();

		expect(component).toMatchSnapshot();
	});

	it('should render with a message and a title', () => {
		component = new LoadingModal({
			message: 'foo',
			title: 'bar'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render an icon', () => {
		component = new LoadingModal({
			icon: 'foo'
		});

		expect(component).toMatchSnapshot();
	});
});