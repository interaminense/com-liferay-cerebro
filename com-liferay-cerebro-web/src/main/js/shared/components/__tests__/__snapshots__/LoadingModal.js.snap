// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal should render 1`] = `
<div class="modal-dialog modal-sm loading-modal-root">
  <div class="modal-content">
    <div class="icon-container">
      <div class="spinner-root loading-icon dotted loading-icon-lg"></div>
    </div>
    <p class="message">loading</p>
  </div>
</div>
`;

exports[`LoadingModal should render an icon 1`] = `
<div class="modal-dialog modal-sm loading-modal-root">
  <div class="modal-content">
    <div class="icon-container">
      <svg class="icon-root lexicon-icon lexicon-icon-foo icon-size-xl">
        <use xlink:href="/images/lexicon/icons.svg#foo"></use>
      </svg>
    </div>
    <p class="message">loading</p>
  </div>
</div>
`;

exports[`LoadingModal should render with a message and a title 1`] = `
<div class="modal-dialog modal-sm loading-modal-root">
  <div class="modal-content">
    <h1 class="title">bar</h1>
    <div class="icon-container">
      <div class="spinner-root loading-icon dotted loading-icon-lg"></div>
    </div>
    <p class="message">foo</p>
  </div>
</div>
`;
